cameo/strain_design/deterministic/linear_programming.py

```diff
@@ -426,3 +430,394 @@ def _repr_html_(self):
+            # problem.parameters.timelimit.set(200) # Use max 200 seconds for solving
+            # problem.parameters.mip.limits.populate.set(20) # Find max 20 solutions (=default)
+
+        else:
+            logger.warning("You are trying to run 'GrowthCouplingPotential' with %s. This might not end well." %
+                           model.solver.interface.__name__.split(".")[-1])
+
+        # Remove reactions that are blocked: no flux through these reactions possible.
+        # This will reduce the search space for the solver, if not done already.
+        if remove_blocked:
+            blocked_reactions = cameo.flux_analysis.analysis.find_blocked_reactions(model)
+            model.remove_reactions(blocked_reactions)
+            blocked_reaction_ids = [r.id for r in blocked_reactions]
+            knockout_reactions = [r for r in knockout_reactions if r not in blocked_reaction_ids]
+            knockin_reactions = [r for r in knockin_reactions if r not in blocked_reaction_ids]
+            medium_additions = [r for r in medium_additions if r not in blocked_reaction_ids]
+            logger.debug("Removed " + str(len(blocked_reactions)) + " reactions that were blocked")
+
+        # Make dual
+        copied_model = model.copy()
+        copied_model.optimize()
+        dual_problem = convert_linear_problem_to_dual(copied_model.solver)
+        logger.debug("Dual problem successfully created")
+
+        # Combine primal and dual
+        primal_problem = model.solver
+
+        for var in dual_problem.variables:  # All variables in the dual are copied to the primal
+            var = primal_problem.interface.Variable.clone(var)
+            primal_problem.add(var)
+        for const in dual_problem.constraints:  # All constraints in the dual are copied to the primal
+            const = primal_problem.interface.Constraint.clone(const, model=primal_problem)
+            primal_problem.add(const)
+        logger.debug("Dual and primal combined")
+
+        dual_problem.optimize()
+
+        # Dictionaries to hold the binary control variables:
+        native_y_vars = {}  # 1 for knockout, 0 for active
+        heterologous_y_vars = {}  # 1 for knockin, 0 for inactive
+        medium_y_vars = {}  # 1 for medium addition (up to -10), 0 for no addition
+
+        # Now the fun stuff
+        constrained_dual_vars = set()
+
+        # Fill the dictionaries with binary variables
+        # For the knockouts: only for the native reactions which are not exchanges
+        # for reaction in [r for r in model.reactions - excluded_reactions if r.type == "native"]:
+        for reac_id in knockout_reactions:
+            reaction = model.reactions.get_by_id(reac_id)
+
+            # Add constraint variables
+            interface = model.solver.interface
+            y_var = interface.Variable("y_" + reaction.id, type="binary")
+
+            # Constrain the primal: flux through reactions within (-1000, 1000)
```

Comment:
This might depend on `cobra.Configuration.bounds`?

Reply:
Yes, this should be taken into account now